File: coredata/ObjC/_PersonWrapper.m
#import "_PersonWrapper.h"
#import "Person.h"

@interface _PersonWrapper ()
@end

@implementation _PersonWrapper {
}
- (instancetype)initWithEntity:(Person *)entity {
    self = [super init];
    if (self) {
        _entity = entity;
        self.birthday = entity.birthday;
        self.name = entity.name;
    }
    return self;
}

- (void)updateEntity:(Person *)entity {
    entity.birthday = self.birthday;
    entity.name = self.name;
}
@end

File: coredata/Swift/_PersonWrapper.swift
import Foundation

class _PersonWrapper: NSObject {
    let _entity: Person?
    func entity() -> Person? {
        return _entity
    }
    var birthday: NSDate?
    var name: String?

    init(person: Person?) {
        _entity = person
        super.init()
        if let e = person {
            self.birthday = e.birthday
            self.name = e.name
        }
    }
    func updatePerson(person: Person) {
        person.birthday = self.birthday
        person.name = self.name
    }
}
